src/lambda.js

```diff
@@ -0,0 +1,71 @@
+import AWS from 'aws-sdk';
+import uuid from 'uuid';
+
+import logger from './logger';
+import {
+	isObjectEmpty,
+	doesObjectHaveValidExtension,
+	getOrganizationSlug,
+	getOrganizationId,
+	getSrcHead,
+	getDecodedSrcKey,
+	getExtension,
+} from './utils';
+import {
+	SOURCE_BUCKET,
+	DESTINATION_BUCKET,
+} from './constants';
+
+const s3 = new AWS.S3();
+
+exports.handler = async (event, context, callback) => {
+	const {
+		s3: {
+			object,
+		},
+	} = event.Records[0];
+	logger.info(`Processing object: ${object.key}`);
+
+	const srcKey = getDecodedSrcKey(object);
+	const extension = getExtension(object);
+	const organizationSlug = getOrganizationSlug(object);
+
+	if (isObjectEmpty(object)) {
+		logger.info('Cannot process upload (empty file)');
+		return callback(null, '0 Size');
+	}
+	if (!doesObjectHaveValidExtension(object)) {
+		logger.info(`Cannot process upload (invalid extension: ${extension})`);
+		return callback(null, 'Invalid Extension');
+	}
+
+	const organizationId = await getOrganizationId(organizationSlug);
```

Comment:
One thing I get nervous about (perhaps because I just don’t understand well enough) is error catching on async/await code. With promises, it’s easy to check that every chain has a `.catch()` at the end. But async/awaits need to be wrapped in try/catches. Is there something there that will catch (and log) errors of that flavor?

Reply:
I did some poking around with async / await and [error handling](https://javascript.info/async-await#error-handling)

Here's the short story:

1. A rejected promise from an await clause is the equivalent of a `throw Error()`.
2. Any error inside of an async function results in a rejected promise.

So for example:

```
const f = async () => throw new Error('Fly you fools')

// This works
f().catch((e) => { ... })

// So does this
try {
  await f()
} catch (e) { ... }
```

In short: async functions can be treated like promises at the top level, and await clauses should be treated like any function that could throw an error.

Now in our case, since the Lambda handler is async in our case, I'm not entirely sure how Amazon will handle a rejected promise in the event of an error.  That said I will add some try / catches to make sure it doesn't reject in the first place.

Reply:
Resolved this by wrapping the await clauses in try / catch blocks.  The result is almost the same (an error results in an error), but I think this is a bit more explicit and we get the opportunity to log that warning.